webapp/app/elster_client/elster_client.py

```diff
@@ -212,12 +212,16 @@ def _generate_est_request_data(form_data, year=2020):
     if adapted_form_data.get('steuernummer_exists') == 'no' and adapted_form_data.get('request_new_tax_number'):
         adapted_form_data['submission_without_tax_nr'] = True
 
-    if not current_user.is_active:
-        # no non-active user should come until here, but they should certainly not be able to send a tax
+    digitally_signed = current_user.is_authenticated and current_user.is_active
+
+    if not digitally_signed:
+        logger.warn('Elster_Client: User is not authenticated or active.')  
+        # no non-active user should come until here, but they should certainly not be able to send a tax        
         logout_user()
+        
     meta_data = {
         'year': year,
-        'is_digitally_signed': current_user.is_active
+        'is_digitally_signed': digitally_signed
```

Comment:
See my review comment for more, but here's how I think the code _should_ look.

```suggestion
meta_data = {
        'year': year,
        'is_digitally_signed': True
```

Reply:
Imo we should (at least for now) not set `is_digitally_signed` always to True. We changed this on purpose from setting it always to True because of the following reasons:

Currently, my understanding of `is_digitally_signed` is that we somehow ensure that the tax declaration can be digitally signed. We only want to do this for users with authenticated identification information. These are all the users who with an FSC. If we just set it to `True` and we change something at some point (e.g. allowing inactive users to log in) we are only relying on the route protection for conveying this critical information and we do not know why it sometimes fails. Because of that, I would not feel comfortable to change it and then send some tax information to Erica.

That said, we might be able to get rid of this with the new Eric version anyway because it is getting rid of the digitally_signed attribute and we then might have to discuss, if we need to send a boolean to Erica to indicate if the tax declaration is okay. I would then rather think about having a guard in the elster_client to only send the information for verified users

Reply:
We could also check for the current user's FSC to make it explicit

Reply:
I agree that checking for the user's FSC would be more explicit, so let's do that.

I still feel pretty strongly that the elster_service should not be concerned with user objects at all and it should _definitely_ not have any dependency on flask-login (and transitively the flask session).

If we want to keep this this check, my suggestion would be to move it up the call stack, probably into the relevant Step.

Happy to jump into a huddle on this with both of you, too.